scripts/override_images/override_release_images.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Overrides the list of release images according to
+OPENSHIFT_VERSION or OPENSHIFT_INSTALL_RELEASE_IMAGE environment variables
+"""
+import json
+import os
+import sys
+from argparse import ArgumentParser
+
+import semver
+from assisted_service_client.models import ReleaseImage
+
+from assisted_test_infra.test_infra import utils
+from consts import DEFAULT_CPU_ARCHITECTURE
+
+
+def get_release_image(
+    release_images: list[ReleaseImage], ocp_xy_version: str, cpu_architecture: str = DEFAULT_CPU_ARCHITECTURE
+) -> ReleaseImage | None:
+    archs_images = [v for v in release_images if v.cpu_architecture == cpu_architecture]
+    release_image = [v for v in archs_images if v.openshift_version == ocp_xy_version]
+    if len(release_image) >= 1:
+        return release_image[0]
+
+    return None
+
+
+def main():
+    # Load default release images
+    with open(args.src, "r") as f:
+        release_images_parsed: list = json.load(f)
+        release_images = [ReleaseImage(**release_image_dict) for release_image_dict in release_images_parsed]
+
+    if (release_image_ref := os.getenv("OPENSHIFT_INSTALL_RELEASE_IMAGE", "")) != "":
+        ocp_semantic_version = utils.extract_version(release_image=release_image_ref)
+        ocp_xyz_version = f"{ocp_semantic_version.major}.{ocp_semantic_version.minor}.{ocp_semantic_version.patch}"
+        ocp_xy_version = f"{ocp_semantic_version.major}.{ocp_semantic_version.minor}"
+
+        if (release_image := get_release_image(release_images=release_images, ocp_xy_version=ocp_xy_version)) is None:
+            release_image = ReleaseImage(
+                openshift_version=ocp_xy_version,
+                version=ocp_xyz_version,
+                cpu_architecture=DEFAULT_CPU_ARCHITECTURE,
+                url=release_image_ref,
+            )
+
+    elif (version := os.getenv("OPENSHIFT_VERSION", "")) != "":
```

Comment:
is it a CLI command line or a SCRIPT? 
please move it to a arg and set it as env default

Reply:
Why do we need to pass it as arg if it exists as env var ?

Reply:
resolved